aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+  :lines: 30-36
+  :emphasize-lines: 4
+  :dedent: 8
+
+Alternately, you can add just the output formatter:
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.OutputFormatters.Add(new XmlSerializerOutputFormatter());
+  });
+
+Once you've added support for XML formatting, your API methods should return the appropriate format based on the request's ``Accept`` header, as this Fiddler example demonstrates:
+
+.. image:: formatting/_static/xml-response.png
+
+You can see in the Inspectors tab that the Raw GET request was made with an ``Accept: application/xml`` header set. The response pane shows the ``Content-Type: application/xml`` header, and the ``Author`` object has been serialized to XML.
+
+Use the Composer tab to modify the request to specify ``application/json`` in the ``Accept`` header. Execute the request, and the response will be formatted as JSON:
+
+.. image:: formatting/_static/json-response-fiddler.png
+
+In this screenshot, you can see the request sets a header of ``Accept: application/json`` and the response specifies the same as its ``Content-Type``. The ``Author`` object is shown in the body of the response, in JSON format.
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. The ``[Produces]`` filter specifies a response format. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope.
+
+.. code-block:: c#
+
+  [Produces("application/json")]
+  public class AuthorsController
+
+The ``[Produces]`` filter will force all actions within the ``AuthorsController`` to return JSON-formatted responses, even if other formatters were configured for the application and the client provided an ``Accept`` header requesting a different, available format. See :doc:`/mvc/controllers/filters` to learn more, including how to apply filters globally.
+
+Special Case Formatters
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Some special cases are implemented using built-in formatters. By default, ``string`` return types will be formatted as `text/plain` (`text/html` if requested via ``Accept`` header). This behavior can be removed by removing the ``TextOutputFormatter``. You remove formatters in the ``Configure`` method in *Startup.cs* (shown below). Actions that have a model object return type will return a 204 No Content response when returning ``null``. This behavior can be removed by removing the ``HttpNoContentOutputFormatter``. The following code removes the ``TextOutputFormatter`` and `HttpNoContentOutputFormatter``.
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.OutputFormatters.RemoveType<TextOutputFormatter>();
+    options.OutputFormatters.RemoveType<HttpNoContentOutputFormatter>();
+  });
+  
+Without the ``TextOutputFormatter``, ``string`` return types return 406 Not Acceptable.
```

Comment:
I've confirmed the behavior is as written. JSON will not handle string. XML, if configured, will (if TextOutputFormatter is removed).